tests/wait-for-images.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Playwright has no built-in way to wait for all on-page images
+ * to be flagged as completed, so we have our own function for that.
+ * @param {Playwrite Page} page
+ * @returns resolve() once all images are done, or reject() if after 20 tries images still haven't finished.
+ */
+module.exports = async function waitForImagesToLoad(page) {
+  page.on(`console`, console.log);
+
+  const images = page.locator("img");
+
+  return new Promise(async (resolve, reject) => {
+    let cutoff = 20;
+    (async function testLoaded() {
+      // force-load all lazy content
+      await images.evaluateAll((imgs) => {
+        const visible = Array.from(imgs).filter(i => i.offsetParent !== null)
```

Comment:
If an element isn't visible due to `display:none` anywhere on it or its ancestors, `e.offsetParent` will be null